posthog/api/test/test_feature_flag.py

```diff
@@ -812,6 +812,43 @@ def test_updating_a_feature_flag_with_same_team_and_key_of_a_deleted_one(self):
         instance.refresh_from_db()
         self.assertEqual(instance.key, "alpha-feature")
 
+    def test_my_flags_is_not_nplus1(self):
+
+        query_counts: List[int] = []
+        queries: List[List[Dict[str, str]]] = []
+
+        count, qs = self._get_flags_counting_queries()
+        query_counts.append(count)
+        queries.append(qs)
+
+        # add new flags
+        for i in range(5):
+            self.client.post(
+                f"/api/projects/{self.team.id}/feature_flags/",
+                data={"name": f"flag-{i}", "key": f"flag-{i}", "filters": {"groups": [{"rollout_percentage": i,}]},},
+                format="json",
+            ).json()
+
+            count, qs = self._get_flags_counting_queries()
+            query_counts.append(count)
+            queries.append(qs)
+
+        # query counts don't climb as flags are added
```

Comment:
IMO being clever (looping, etc) in tests is often a smell - e.g. this test is hard to understand what's going on.

Suggestion:
- Create N (e.g. 4) flags:
- Call (and verify) query counts once.